packages/cli/src/care/CarePackage.ts

```diff
@@ -13,66 +13,105 @@ const ELEMENT_TYPE_DEFAULTS: any = {
         files: "*.zip"
     }
 }
-export interface Element {
-    type: ElementType,
+export interface ElementSource extends Element {
     base: string,
     outDir: string,
-    files: string
+    files: string,
+}
+
+export interface Element {
+    type: ElementType,
+    location: string,
+}
+
+export interface CarePackageSourceSpec {
+    name?: string,
+    version?: string,
+    vendor: string,
+    specVersion: string,
+    elements: ElementSource[]
 }
 
 export interface CarePackageSpec {
+    name: string,
+    version: string,
+    vendor: string,
+    specVersion: string,
     elements: Element[]
 }
 
-export class CarePackage {
-    private spec: CarePackageSpec
+export class CarePackageSource {
+    private spec: CarePackageSourceSpec
 
-    private constructor(public packageRoot: string, spec: any) {
+    constructor(public packageRoot: string, spec: any) {
         this.spec = {
             ...spec,
             elements: spec.elements.map((ele: any) => {
```

Comment:
Isn't this from type Element?

Reply:
Well its not since `outDir` and `files` are optional, where as in the `ElementSource` they are not. I could potentially introduce a new type, but I don't see the need for it, as it would only indicate what I expect, without ability to verify it, since the data is loaded from JSON.